source/game/kart/KartCollide.cc

```diff
@@ -360,6 +363,15 @@ void KartCollide::processWheel(CollisionData &collisionData, Hitbox &hitbox,
 void KartCollide::processBody(CollisionData &collisionData, Hitbox &hitbox,
         Field::CourseColMgr::CollisionInfo *colInfo, Field::KCLTypeMask *maskOut) {
     processFloor(collisionData, hitbox, colInfo, maskOut, false);
+    const auto colDirector = Field::CollisionDirector::Instance();
+    bool hitCannon =
+            colDirector->findClosestCollisionEntry(maskOut, KCL_TYPE_BIT(COL_TYPE_CANNON_TRIGGER));
+    if (hitCannon) {
+        // cannon's kcl variant is the cannon point id
```

Comment:
@vabold Feel free to intervene here, but I think this comment can be removed. The purpose of the setter is already clear, and the value being passed into the setter is also clear already.
```suggestion
```